crates/core/src/network/networking.rs

```diff
@@ -82,14 +80,16 @@ impl NetworkSender for Networking {
             };
 
             attempts += 1;
-            if attempts > DEFAULT_MAX_CONNECT_ATTEMPTS {
-                return Err(NetworkError::Unavailable(format!(
-                    "failed to connect to node {} after {} attempts",
-                    to, DEFAULT_MAX_CONNECT_ATTEMPTS
-                )));
-            }
             if attempts > 1 {
-                sleep_with_jitter(SEND_RETRY_BASE_DURATION).await;
+                if let Some(next_retry_interval) = retry_policy.next() {
+                    tokio::time::sleep(next_retry_interval).await;
```

Comment:
We might need to trace! log here to observe when retries are happening, what do you think?

Reply:
That does seem useful indeed! Will add it.

Reply:
Done; I played around with modifying the existing info! messages but left it as-is in the end. Looks something like this now:

```
  2024-08-01T12:30:26.052676Z TRACE restate_types::retries::tests: Delaying retry, attempt: 1, delay: 25.144955ms
    at crates/types/src/retries.rs:345 on retries::tests::test_print_attempts

  2024-08-01T12:30:26.052688Z TRACE restate_types::retries::tests: Delaying retry, attempt: 2, delay: 51.13405ms
    at crates/types/src/retries.rs:345 on retries::tests::test_print_attempts

  2024-08-01T12:30:26.052696Z TRACE restate_types::retries::tests: Delaying retry, attempt: 3, delay: 88.22924ms
    at crates/types/src/retries.rs:345 on retries::tests::test_print_attempts
```